src/main/java/de/tum/in/www1/artemis/web/rest/TeamResource.java

```diff
@@ -238,16 +239,17 @@ public TeamResource(TeamRepository teamRepository, TeamService teamService, Cour
     }
 
     /**
-     * GET /teams?shortName={shortName} : get boolean flag whether team with shortName exists
+     * GET /exercises/{exerciseId}/teams/exists?shortName={shortName} : get boolean flag whether team with shortName exists for exercise
      *
+     * @param exerciseId the id of the exercise for which to check teams
      * @param shortName the shortName of the team to check for existence
      * @return Response with status 200 (OK) and boolean flag in the body
      */
-    @GetMapping("/teams") // this check is independent of the exercise (a team's shortName serves as a globally unique identifier)
+    @GetMapping("/exercises/{exerciseId}/teams/exists")
     @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
-    public ResponseEntity<Boolean> existsTeamByShortName(@RequestParam("shortName") String shortName) {
-        log.debug("REST request to check Team existence by shortName : {}", shortName);
-        return ResponseEntity.ok().body(teamRepository.findOneByShortName(shortName).isPresent());
+    public ResponseEntity<Boolean> existsTeamByShortName(@PathVariable long exerciseId, @RequestParam("shortName") String shortName) {
```

Comment:
There is no check in place here, that the TA/Instructor is TA/Instructor of that given course, right?
E.g.
```
        if (!authCheckService.isAtLeastInstructorForExercise(destinationExercise, user)) {
            return forbidden();
        }
```

Reply:
Good finding! The reason why this check was not there previously was that this was a global existence query that returned whether this team short name exists among all exercises. But now that we have eased the unique constraint and the check has become exercise-specific, we can indeed add the authority check you proposed.